packages/api/src/platforms/vtex/utils/canonical.ts

```diff
@@ -0,0 +1,10 @@
+import type { Product } from '../clients/search/types/ProductSearchResult'
+
+export const canonicalFromProduct = ({ items, linkText }: Product) => {
+  // Sort by id so we always use the same canonical regardless of sku availability/ordering
+  const [{ itemId }] = items.sort(
```

Comment:
can u exemplify this with a real use case?

Reply:
Search might not follow `itemId` order. So on Store Framework, the indexed SKU would be eg. id 2, but here, it'd be id 1, what could make the product name and other attributes change when users are redirected from `/slug/p` (previously implicit skuId 2) to `/slug-${skuId}/p` (explicit skuId 1).

Reply:
Yes, the reasoning behind this was: 
Suppose you have two skus, A and B. Let's say today A is available and B is out of stock. This means, that, when we request for the product, the items array will be `[A, B]`. This makes the canonical be `/slug-A/p`. Now, suppose that A gets out of stock and B enters the stock. Now, when requesting for the product, the items array would be: `[B,A]`, making the canonical be `/slug-B/p`. 
This sort prevents this behavior making the canonical always be consistently `/slug-A/p`. 

The tradeoff in here is redirecting the user to an out of stock sku vs changing the canonical of a page. I'm not sure which approach is worse for conversion (breaking SEO vs redirecting the user to an out of stock product)

Reply:
yeah, I think redirecting to the same consistent page sounds better to me, even if that means it'll be out of stock. But isn't that a search config? Can't we force the items to be returned in the same order, regardless of availability? I think that's the best scenario.

Reply:
ok, I'll remove the sort then

Reply:
I don't think that's what I meant, Gimenes. Does this mean that, in your example, because of the availability status, another sku can take the first position?